lumicks/pylake/detail/confocal.py

```diff
@@ -6,22 +6,38 @@
 from .mixin import PhotonCounts
 from .mixin import ExcitationLaserPower
 from .image import reconstruct_image_sum, reconstruct_image, save_tiff, ImageMetadata
+from .utilities import could_sum_overflow
+
+
+def _int_mean(a, total_size, axis):
+    if not could_sum_overflow(a, axis):
+        return np.sum(a, axis) // total_size
+
+    n = a.size // 2 if axis is None else a.shape[axis] // 2
+    b = a if axis is None else np.swapaxes(a, axis, 0)
+    return _int_mean(b[:n], total_size, 0) + _int_mean(b[n:], total_size, 0)
+
+
+def timestamp_mean(a, axis=None):
+    minimum = np.min(a)
+    return minimum + _int_mean(a - minimum, a.size if axis is None else a.shape[axis], axis)
```

Comment:
Similarly, it would be good to leave a comment here explaining why we're subtracting the minimum.